lib/rubocop/cop/lint/number_convertion.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for usage of #to_i.
+      #
+      # @example
+      #   # bad
+      #   foo.to_i
+      #
+      #   # good
+      #   Integer(foo)
+      class NumberConvertion < Cop
+        MSG = 'Use `Integer()` instead of `#to_i`.'.freeze
```

Comment:
This message should include why we should use `Integer()` method.

Reply:
Should it though? 🤔 I don't think the other cops encapsulate the "why", which can be quite involved, in the offense message.

Reply:
If user receives this message, user probably confuses. Because it does not mention "why". So I think the message should mention "why". And I think other Lint cops messages should have "why" for the same reason.

`Lint/AmbiguousOperator`'s message is a good example.
https://github.com/bbatsov/rubocop/blob/4a0d6361d0065ca16ba19bdf3b3d6c4623e14adc/lib/rubocop/cop/lint/ambiguous_operator.rb#L36-L39
It is a bit long, but it is clear.